### ISSUE TITLE
Move .x files to their own repo

### ISSUE THREAD

**Issue by jedmccaleb** [stellar/stellar-core#984]
This will make it easier for other projects to stay in sync with any xdr changes we make.

**Comment by MonsieurNicolas** [comment]
Not arguing necessarily to not do this, but how is that making things easier to stay in sync? Would you need to move the SQL schema as well?

I thought one of the main painpoints wrt .x files being revved is knowing when they get deployed?

Also this could be a good time to split x files into categories (different folders maybe?):
some are ledger related (anything Stellar)
some are meta data related (ie "API" with integrators via SQL, which also includes the SQL schema)
some are p2p protocol related (alternate implementations need to implement these to a certain extent)

**Comment by raymens** [comment]
It's easier to spot any changes if there's a repo for only XDR files. Using git submodules these can be updated and included in other projects. At the same time it's easier to know what version of XDR files are being used by other projects.

**Comment by bartekn** [comment]
I vote for a separate repo that will contain XDR files. Pros:
* The main reason is what @raymens said: it will be very easy to track changes (minor/breaking with semver versioning).
* We can use CI to generate language specific classes from .x files using [xdrgen](https://github.com/stellar/xdrgen) and then use the new XDR repo as a dependency for our SDKs:
  * For example: for JS, we can generate [xdr_generated.js](https://github.com/stellar/js-stellar-base/blob/master/src/generated/stellar-xdr_generated.js) file in `js` folder and create a `package.json` file to export all structures. After a successful build the repo can be published to `npm`.
  * Some dependent repositories (I'm thinking about [xdr-viewer](https://github.com/stellar/xdr-viewer)) will be very easy to update (if we configure it to use the latest version of XDR files).
  * In other words this repo can act as a single multi-language dependency.
* It will make our dependency graph much cleaner. Right now, all SDKs depend on stellar-core because of .x files. After extracting them to a separate repository it will be a single dependency for most of them because SDKs do not communicate with stellar-core directly (they use horizon instead).
* It would be much easier to build helper apps/scripts that use our dependency graph. For example, right now all libs (js, ruby, go, java, etc.) depend on `stellar-core` repository (because of .x files). If we wanted to build a bot that will list of all dependent libraries as a comment under a PR, it would list all SDKs on every stellar-core PR.

**Comment by irisli** [comment]
I think having them in a separate repo would remedy painpoint of the knowing when the .x files get deployed.

That repo could have very clear tagged versions with notes on when the planned update is. It would be lighter to navigate than seeing it in addition to the stellar-core project files.

**Comment by MonsieurNicolas** [comment]
marking as protocol so that we can track this as part of that workstream

**Comment by jedmccaleb** [comment]
This wouldn't be a protocol change though?

**Comment by MonsieurNicolas** [comment]
it's not but it would require a bunch of changes from all downstream repos, just like a protocol change

**Comment by bartekn** [comment]
It seems this happened recently so probably can be closed.

**Comment by MonsieurNicolas** [comment]
only "v-next" was moved - we vendor the current version of the protocol in the core repo and we don't necessarily point core's vnext to `master` so I guess it depends on what people are trying to do (in the case of Horizon, you need a very specific version of core for example)

**Comment by MonsieurNicolas** [comment]
both are now moved
